src/test/java/array/Array1dTest.java

```diff
@@ -0,0 +1,207 @@
+package array;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+
+import java.util.Optional;
+import java.util.stream.IntStream;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class Array1dTest {
+
+    @Test
+    void shouldCreateEmptyArrayByConstructor() {
+        // given
+        int size = 10;
+        // when
+        Array1d<Integer> array = new Array1d<>(size);
+        // then
+        assertNotNull(array);
+    }
+
+    @Test
+    void shouldCreateEmptyArrayByFactoryMethod() {
+        // given
+        int size = 10;
+        // when
+        Array1d<Integer> array = Array1d.of(size);
+        // then
+        assertNotNull(array);
+    }
+
+    @Test
+    void shouldVisitEmptyArray() {
+        // given
+        int size = 10;
+        Array1d<Integer> array = Array1d.of(size);
+        // expect
+        assertDoesNotThrow(() -> array.visit(i -> System.out.print(i + ", ")));
+    }
+
+    @Test
+    void shouldFillArrayWithGivenNumbers() {
+        // given
+        int start = 0;
+        int end = 9;
+        int size = 10;
+        Array1d<Integer> array = Array1d.of(size);
+        // expect
+        assertDoesNotThrow(() ->
```

Comment:
may be verification should be like this
1. fill the array
2. compare it with expected etalon?

Reply:
How to retrieve the state then? Use `equals` or `at` method?
If `equals` should I implement it?
If `at`, for me, it's not isolated testing 'cause we call 2 array methods instead of 1.

Reply:
@PaulFrmBrn please answer the questions.

Reply:
Sorry - didn't see this conversation in github interface

I'd use `visit` method to get state of the array in some convenient form.

Yes, it uses two methods of the SUT, and if `visit` implementation is broken than two test will fail. 

But it's better then situation when If `insert` is broken but throws no error this test is green.

Reply:
Fixed 9a362ee10c6f88166f859e06cfc0775f992d92f8